### DISCUSSION TITLE
Example usage with non-packaged data

### DISCUSSION THREAD

**Discussion by fabianegli** [fgcz/prolfqua#31] [Ideas]
I was looking for the right way to read in MSFragger search results and didn't really find a nice example in the tutorials that showed how to read in such data. Did I just not find it, or is this indeed missing from the documentation/tutorials?

I think a section in the docs that explains how to read in data and examples for all the popular search engines would be really nice.

**Comment by wolski** [comment]
Dear Fabian, 

please do have a look here:
https://wolski.github.io/prolfquabenchmark/articles/BenchmarkMSFraggerProteinIonStar.html

There is an example of reading a "combined_protein.tsv".
If your `combined_protein.tsv` was generated using an MSFragger version 16 or newer then please use the function:
`prolfqua::tidy_MSFragger_combined_protein_V16` to convert the file into a tidy table.

There is also an example of reading MSFragger data here:
https://www.biorxiv.org/content/10.1101/2022.06.07.494524v1
(see supplement)

Hope this helps.

**Reply by fabianegli** [reply]
Thank you for the links. That's indeed very helpful.

It seems that there are functions that are not showing up in the manual: https://rdrr.io/github/wolski/prolfqua/man/

There are many documented `tidy_MQ...`, but no `tidy_MSFragger...` functions listed. Is that intentional?

**Reply by fabianegli** [reply]
Also, I find it a bit strange to see in the following line in the Example of [MSFragger](https://rdrr.io/github/wolski/prolfqua/man/MSFragger.html) because this might imply that the data used is from MaxQuant instead of MSFragger.

```
unzip(inputMQfile, list = TRUE)$Name
```

Also, the linked docs above are reached by clicking MSFragger in the main documentation website, but then there is no function MSFragger, but a couple of functions with different names. Maybe this is just the way R documentation works, but this is somewhat confusing for me. I am also kind of expecting that there must be a function named `tidy_MSFragger_combined_peptides_V15`, but this function is never mentioned. Then I search the docs, find the source where `tidy_MSFragger_combined_peptides_V16` lives and there's a function `[tidy_MSFragger_combined_protein_V16_deprec](https://rdrr.io/github/wolski/prolfqua/src/R/tidyMSFragger.R)`. I have to say I am a bit confused now. Also, because MSFragger is now at version 3.5 (probably a long way from version 16), but FragPipe is at 1.8 which would make more sense if 16 corresponds with the version 1.6 of FragPipe?

I hope this feedback helps to improve the docs and maybe even the function names.

**Comment by wolski** [comment]
Thanks for the comments. Yes, 16 refers to version 1.6 of Fragpipe, actually IonQuant, AFAIK, and not just MSFragger. 
The file formats were relatively stable for a while; then, they changed them with version 1.6. Therefore there is V16 and no version V15 or V14 of the functions. Are you using Fragpipe 1.8? I will have to check if there are no changes yet again to the Fragpipe output formats. They are a bit of a moving target. 

However, the functions should show up in the help; they are not internal:
![image](https://user-images.githubusercontent.com/1926513/174048594-ffd4b843-676e-4364-8352-e26786f5f289.png)
Indeed the documentation title should be better. 
I started to use the @rdname
https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html
recently to document similar functions in one file and do not have much experience yet.

I will update the names of the functions and the documentation for the package's next release (maybe try to autodetect the version of fragpipe used) and then close the ticket.
- tidy_FragPipe_combined_proteins 
- tidy_FragPipe_combined_peptides

Thanks again. Feedback welcome.

**Reply by fabianegli** [reply]
Thank you for the clarifications. I see now, my lack of experience searching for R documentation is also partially to blame. The functions show up in the [Functions](https://rdrr.io/github/wolski/prolfqua/api/) section of the documentation, but not in the list of [Man pages](https://rdrr.io/github/wolski/prolfqua/man/). The MaxQuant equivalent, however, do.

I was using FragPipe version 17.1. It would be great to be able to use prolfqua with these searches.

**Comment by wolski** [comment]
I am sure you can use tidy_MSFragger_combined_protein_V16 with 1.7.1 combined_protein.tsv

**Reply by fabianegli** [reply]
Thank you!

**Comment by wolski** [comment]
Hi Fabian,

I created a new release 
https://github.com/fgcz/prolfqua/releases/tag/v0.4.2

Where I did rename the functions tidy_MSFragger.. to tidy_FragPipe..

https://fgcz.github.io/prolfqua/reference/FragPipe.html

The read combined_protein functions for older versions of fragpipe are now called:
tidy_FragPipe_combined_protein_deprec

All the `combined_protein.tsv` files starting with FragPipE 1.6 can be read using:
tidy_FragPipe_combined_protein
